### PR TITLE
[Workflow]Fix Marking when it must contains more than one tokens

### LINKED ISSUE #53179: [Workflow] Verify race condition
### Symfony version(s) affected

all

### Description

The behavior of transitions depends on the order in which they are declared.

### How to reproduce

```php
    private static function createWorkflowWithSameNameBackTransition(): Definition
    {
        $places = range('a', 'c');

        $transitions = [];
        $transitions[] = new Transition('a_to_bc', 'a', ['b', 'c']);
        $transitions[] = new Transition('back1', 'b', 'a');
        $transitions[] = new Transition('back1', 'c', 'b');
        $transitions[] = new Transition('back2', 'c', 'b');
        $transitions[] = new Transition('back2', 'b', 'a');

        return new Definition($places, $transitions);

        // The graph looks like:
        //       +--------+
        //   +---| back_N |<------+ +-----------------+
        //   |   +--------+       | |                 |
        //   v                    | v                 |
        // +---+    +---------+  +---+  +---+     +--------+
        // | a | -> | a_to_bc |  | b |  | c | --> | back_N |
        // +---+    +---------+  +---+  +---+     +--------+
        //                |        ^      ^
        //                +--------+------+
    }

    public static function backVariants()
    {
        return [['back1'], ['back2']];
    }

    /**
     * @dataProvider backVariants
     */
    public function testApplyWithSameNameBackTransition($transition)
    {
        $subject = new Subject();
        $definition = $this->createWorkflowWithSameNameBackTransition();
        $workflow = new Workflow($definition, new MethodMarkingStore());

        $marking = $workflow->apply($subject, 'a_to_bc');

        $this->assertFalse($marking->has('a'));
        $this->assertTrue($marking->has('b'));
        $this->assertTrue($marking->has('c'));

        $marking = $workflow->apply($subject, $transition);

        $this->assertTrue($marking->has('a'));
        $this->assertTrue($marking->has('b'));
        $this->assertFalse($marking->has('c'));
    }

```

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #53179
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Workflow/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ CHANGELOG
  * Require explicit argument when calling `Definition::setInitialPlaces()`
  * `GuardEvent::getContext()` method has been removed. Method was not supposed to be called within guard event listeners as it always returned an empty array anyway.
  * Remove `GuardEvent::getContext()` method without replacement
+ * Add partial support for marking with more than one token per place
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/CHANGELOG.md]: the word "partial" means nothing to the reader i think (it doesn't to me at least)

**Review by nicolas-grekas — COMMENTED** [review]
the word "partial" means nothing to the reader i think (it doesn't to me at least) :)

File: src/Symfony/Component/Workflow/CHANGELOG.md (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/CHANGELOG.md]: I don't know how to said that shortly but

There was an issue with the previous version of the workflow. see #53179. This PR fix this case, but open the door to real multi token per place. 

In another PR (much more bigger) I planned to add a real support. Symfony will expose a `weight` attribute on each transition, and this value will be used during the `apply` and `can` method.

Comment (stof) [reply src/Symfony/Component/Workflow/CHANGELOG.md]: Well, the changelog should describe what is supported. Saying `partial support` does not explain what gets added (maybe that's supporting the whole feature set but only during full moon, which would also make it a partial support :trollface:)

Comment (lyrixx) [reply src/Symfony/Component/Workflow/CHANGELOG.md]: Maybe this is a bug fix instead. WDYT ?

**Comment by lyrixx** [comment]
I updated the PR. Now it's a bugfix, and not a new feature

File: src/Symfony/Component/Workflow/Marking.php [hunk 2]
```diff
@@ -19,31 +19,54 @@
...
+        }
+
+        if (!\array_key_exists($place, $this->places)) {
+            $this->places[$place] = 0;
+        }
+        $this->places[$place] += $nbToken;
     }
 
-    public function unmark(string $place)
+    public function unmark(string $place, int $nbToken = 1)
```
Comment (chalasr) [inline src/Symfony/Component/Workflow/Marking.php:46]: this change is a BC break for child classes overriding the method with extra parameter(s). It should rather be done the smooth way on 7.1

**Review by chalasr — COMMENTED** [review]
Reading the code and the linked issue, I don't think this should have been merged as a bugfix.

**Comment by lyrixx** [comment]
Sorry for this mistake. the PR was open for a long period, so I thought it was okay.

I have reverted the merge, and I'll open a new PR against 7.1 ASAP

**Pull request merged**